### PR TITLE
try to fix https://github.com/fsprojects/Paket/issues/2640 

### PR DESCRIPTION
**Pull request by matthid**
by always calling getVersions.

 https://github.com/fsprojects/Paket/issues/2640

### CODE REVIEW

**Comment by matthid** [comment]
Ok here is the explanation for this:
 - "Assuming" a version as the resolver did for "pinned" version broke my assumption that "GetDetails" cannot fail (because GetVersions already returned a version)
 - Apparently this was a "feature", because it is the only way to install packages which are not indexed yet by NuGet

Therefore, the solution is:
 - We always call GetVersions before
 - If there are NO results (maybe the case where package is not indexed) we save a `bool` to indicate that we "assumed" the version and forward that info to GetDetails
 - GetDetails will no longer blacklist if it sees that the version was "assumed"

I noticed another problem where NuGet sources are asked when the package only existed in a local one, I hope this is solved by this as well (and was only a special case of this).
Will add a test for this case with pinned version.

File: global.json [hunk 1]
```diff
@@ -1,5 +0,0 @@
-{
-  "sdk": {
-    "version": "2.1.0-preview1-007002"
```
Comment (0x53A) [inline global.json]: maybe we should instead lock to 2.0.0 (the release version)? Do you know why this was originally added?

Comment (matthid) [reply global.json]: I have no idea, I guess it was before 2.0.0 was released. yes we should probably  lock and downgrade the fake script.

This was just to get rid of the very annoying blocker locally.

Comment (matthid) [reply global.json]: I have no idea, I guess it was before 2.0.0 was released. yes we should probably  lock and downgrade the fake script.

This was just to get rid of the very annoying blocker locally.

**Pull request merged**